src/core/objective.jl

```diff
@@ -276,16 +276,17 @@ function _objective_min_fuel_and_flow_cost_polynomial_nl(pm::AbstractPowerModel;
 
             cost_rev = reverse(gen["cost"])
             if length(cost_rev) == 1
-                gen_cost[(n,i)] = JuMP.@NLexpression(pm.model, cost_rev[1])
+                gen_cost[(n,i)] = JuMP.@expression(pm.model, cost_rev[1])
             elseif length(cost_rev) == 2
-                gen_cost[(n,i)] = JuMP.@NLexpression(pm.model, cost_rev[1] + cost_rev[2]*pg)
+                gen_cost[(n,i)] = JuMP.@expression(pm.model, cost_rev[1] + cost_rev[2]*pg)
             elseif length(cost_rev) == 3
-                gen_cost[(n,i)] = JuMP.@NLexpression(pm.model, cost_rev[1] + cost_rev[2]*pg + cost_rev[3]*pg^2)
+                gen_cost[(n,i)] = JuMP.@expression(pm.model, cost_rev[1] + cost_rev[2]*pg + cost_rev[3]*pg^2)
             elseif length(cost_rev) >= 4
                 cost_rev_nl = cost_rev[4:end]
-                gen_cost[(n,i)] = JuMP.@NLexpression(pm.model, cost_rev[1] + cost_rev[2]*pg + cost_rev[3]*pg^2 + sum( v*pg^(d+3) for (d,v) in enumerate(cost_rev_nl)) )
+                # TODO(odow): is the correct? Or should it be pg^(d+2.0)?
+                gen_cost[(n,i)] = JuMP.@expression(pm.model, cost_rev[1] + cost_rev[2]*pg + cost_rev[3]*pg^2 + sum( v*pg^(d+3.0) for (d,v) in enumerate(cost_rev_nl)) )

```

Comment:
This looked wrong, because there is no cubic term. It goes from `cost_rev[3]*pg^2 + cost_rev[4]*pg^4`.

Should it just be: 
```julia
gen_cost[(n,i)] = JuMP.@expression(
    pm.model, 
    sum(v * pg^(d - 1) for (d, v) in enumerate(reverse(gen["cost"]))),
)
```

Reply:
Yes this sounds like a bug, I will fix this on the current master release and also develop a test that will catch this issue.